modules/core/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsManager.java

```diff
@@ -28,9 +35,29 @@
      * @param objName Object to collect statistics by.
      * @param colNames Columns to collect statistics by.
      * @throws IgniteCheckedException  Throws in case of errors.
+     * @return future to get fini
      */
     public void collectObjectStatistics(String schemaName, String objName, String... colNames) throws IgniteCheckedException;
 
+    /**
+     * Collect objects statistics.
+     *
+     * @param keys Collection of keys to collect statistics by (schema, obj, columns).
+     * @return Future to track progress and cancel collection.
+     * @throws IgniteCheckedException In case of errors.
+     */
+    public StatsCollectionFuture<Map<GridTuple3<String, String, String[]>, ObjectStatistics>> collectObjectStatisticsAsync(
```

Comment:
I don't think we need a method to collect statistics by several objects at all

Reply:
We discussed it - we need to collect statistics by several tables in the same cache group in single query.